packages/mux-player-react/src/useCombinedRefs.ts

```diff
@@ -13,15 +13,24 @@ export const useCombinedRefs: useCombinedRefs = (...refs) => {
   const targetRef = useRef(null);
 
   useEffect(() => {
+    const cleanupFunctions: (() => void)[] = [];
+
     refs.forEach((ref) => {
       if (!ref) return;
 
       if (typeof ref === 'function') {
-        ref(targetRef.current);
+        const result = ref(targetRef.current);
+        if (typeof result === 'function') {
+          cleanupFunctions.push(result);
+        }
       } else {
         ref.current = targetRef.current;
       }
     });
+
+    return () => {
+      cleanupFunctions.forEach((cleanup) => cleanup());
+    };
   }, [refs]);
 
   return targetRef;
```

Comment:
this looks very good! 

doing some quick search on the internet I did notice some cleanup is missing for the React 18 and under case. 
let's replace this complete function with this implementation 
https://gist.github.com/KurtGokhan/9aafd8e83c9bc6a2946fe2dc7f2c1d19#file-use-combined-refs-new-ts

it takes both the old and new React in account completely.